compiler/test/dotty/tools/scripting/BashExitCodeTests.scala

```diff
@@ -4,187 +4,64 @@ package scripting
-   * @param className name of compiled class
-   * @param expectedRunExitCode expected exit code from the runner
-   */
-  private def compileRunAndVerifyExitCode(
-    testName: String,
-    className: String,
-    expectedRunExitCode: Int,
-  )(using File): Unit =
-    compileAndVerifyExitCode(testName, 0)
-    runClassAndVerifyExitCode(className, expectedRunExitCode)
-
-  /*
-   * Runs the command and checks the exit code
-   *
-   * @param args arguments for command line
-   * @param expectedExitCode expected exit code from the output
-   */
-  private def testCommandExitCode(args: Seq[String], expectedExitCode: Int): Unit =
-    val commandline = args.mkString(" ")
-    val (validTest, exitCode, _, _) = bashCommand(commandline)
-    if verifyValid(validTest) then
-      assertEquals(expectedExitCode, exitCode)
-
-  /*
-   * Checks if scripting test resources contains test with given `testName`
-   * And then runs function `test`
-   *
-   * @param testName name of the test containing the extension
-   * @param test check to be run on found test file
-   */
-  private def assertTestExists(testName: String)(test: File => Unit) =
-    val file = testFiles.find(_.getName == testName)
-    assert(file.isDefined)
-    test(file.get)
-
-  /*
-   * Runs test for created temporary file
-   * and ensures it deletion after function execution
-   *
-   * @param test check to be run on found test file
-   */
-  private def withTempFile(test: File => Unit) =
-    val tempFile = Files.createTempFile("temp-file", ".class").toFile
-    try {
-      test(tempFile)
-    } finally {
-      Util.deleteFile(tempFile)
-    }
-
-  /*
-   * Runs test with implicit temporary directory
-   * and ensures it deletion after the function execution
-   *
-   * @param test test to be run with given temporary directory
-   */
-  private def withTempDirectory(test: File ?=> Unit) =
-    given file: File = Files.createTempDirectory("exit-code-tests").toFile
-    try { test } finally { Util.deleteFile(file) }
-
-  /*
-   * Returns path to the generated tasty file for given directory and classname
-   */
-  private def getGeneratedTastyPath(className: String)(using temporaryDir: File): String =
-    val file = temporaryDir.files.find(_.getName == s"$className.tasty")
-    assert(file.isDefined)
-    file.get.absPath
-
 @Category(Array(classOf[BootstrappedOnlyTests]))
 class BashExitCodeTests:
-  import BashExitCodeTests.*
-
-  @Test def verifyExitCodeOnCompileError: Unit =
-    withTempDirectory(compileAndVerifyExitCode("compileError.scala", 1))
-
-  @Test def verifyExitCodeOnRuntimeError: Unit =
-    withTempDirectory(compileRunAndVerifyExitCode("runtimeError.scala", "runtimeError", 1))
+  private var myTmpDir: String | Null = null
+  private lazy val tmpDir = { myTmpDir = Files.createTempDirectory("exit-code-tests").toFile.absPath; myTmpDir }
+  @After def cleanup(): Unit = if myTmpDir != null then io.Directory(myTmpDir).deleteRecursively()
```

Comment:
It looks like `myTmpDir` is only used to check if temporary directory was initialized.
I think it would be better to just write
```scala
private lazy val tmpDir = Files.createTempDirectory("exit-code-tests").toFile.absPath
@After def cleanup(): Unit = if Files.exists(tmpDir.toPath) then io.Directory(tmpDir).deleteRecursively()
```

Reply:
Your version always creates the directory, even when running `Files.exists`.  I was trying to avoid creating it when it's not needed.

Reply:
Scala 2 uses `Using.resources` with implicits in `testkit.ReleasablePath` for performing the cleanup. I recall Windows always balked at trying to delete something "in use".